modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1463,14 +1466,129 @@ export class Wallet implements IWallet {
     return this.bitgo.post(this.url('/share')).send(params).result();
   }
 
+  /**
+   * Shares a wallet with multiple users by creating bulk wallet shares.
+   *
+   * @async
+   * @param {BulkWalletShareOptions} params - The options for sharing wallets in bulk.
+   * @param {Array<ShareOption>} params.shareOptions - An array of share option objects containing user and permissions information.
+   * @param {Object} [params.shareOptions[].keychain] - The keychain object used to share the wallet.
+   * @param {string} [params.shareOptions[].keychain.toPubKey] - The recipient's public key.
+   * @param {string} [params.shareOptions[].keychain.path] - The derivation path of the keychain.
+   * @param {string} params.shareOptions[].user - The user to share the wallet with.
+   * @param {string} params.shareOptions[].permissions - The permissions granted to the user.
+   * @param {string} [params.walletPassphrase] - The wallet passphrase used to decrypt the keychain.
+   * @throws {Error} If `shareOptions` is empty, or if required keychain parameters (`toPubKey` and `path`) are missing when needed.
+   * @throws {Error} If unable to decrypt the user keychain.
+   * @returns {Promise<CreateBulkWalletShareListResponse>} A promise that resolves with the response of the bulk wallet share creation.
+   */
+  async bulkWalletShare(params: BulkWalletShareOptions): Promise<CreateBulkWalletShareListResponse> {
+    if (!params.shareOptions || Object.keys(params.shareOptions).length === 0) {
+      throw new Error('shareOptions cannot be empty');
+    }
+    const bulkCreateShareOptions: BulkCreateShareOption[] = [];
+
+    for (const shareOption of params.shareOptions) {
+      common.validateParams(shareOption, ['userId', 'pubKey', 'path'], []);
+
+      const needsKeychain = shareOption.permissions && shareOption.permissions.includes('spend');
+      let sharedKeychain: any = undefined;
+
+      if (needsKeychain) {
+        try {
+          const keychain = await this.getEncryptedUserKeychain();
+
+          if (!keychain.encryptedPrv) {
+            sharedKeychain = undefined;
+          } else {
+            const userPrv = decryptKeychainPrivateKey(this.bitgo, keychain, params.walletPassphrase);
+            if (!userPrv) {
+              throw new Error('Unable to decrypt user keychain.');
+            }
+
+            const eckey = makeRandomKey();
```

Comment:
```suggestion
            const ecdhkey = makeRandomKey();
```

Reply:
Are you sure this is how the ecdh key is generated? 🤔 I beleive we have some standardized methods we use to generate it